sdk/testing/stepwise/environments/docker/environment.go

```diff
@@ -0,0 +1,858 @@
+
+	n.container, err = r.Start(context.Background())
+	if err != nil {
+		return err
+	}
+
+	n.Address = &net.TCPAddr{
+		IP:   net.ParseIP(n.container.NetworkSettings.IPAddress),
+		Port: 8200,
+	}
+	ports := n.container.NetworkSettings.NetworkSettingsBase.Ports[nat.Port("8200/tcp")]
+	if len(ports) == 0 {
+		n.Cleanup()
+		return fmt.Errorf("could not find port binding for 8200/tcp")
+	}
+	n.HostPort = ports[0].HostPort
+
+	return nil
+}
+
+// DockerClusterOptions has options for setting up the docker cluster
+type DockerClusterOptions struct {
+	KeepStandbysSealed bool
+	RequireClientAuth  bool
+	SkipInit           bool
+	CACert             []byte
+	NumCores           int
+	tmpDir             string
+	PluginTestBin      string
+	// SetupFunc is called after the cluster is started.
+	SetupFunc func(t testing.T, c *DockerCluster)
+	CAKey     *ecdsa.PrivateKey
+}
+
+//
+// test methods/functions
+//
+
+// testWaitHealthMatches checks health
+func testWaitHealthMatches(ctx context.Context, client *api.Client, ready func(response *api.HealthResponse) error) error {
+	var health *api.HealthResponse
+	var err error
+	for ctx.Err() == nil {
+		health, err = client.Sys().Health()
+		switch {
+		case err != nil:
+		case health == nil:
+			err = fmt.Errorf("nil response to health check")
+		default:
+			err = ready(health)
+			if err == nil {
+				return nil
+			}
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	return fmt.Errorf("error checking health: %v", err)
+}
+
+func testWaitLeaderMatches(ctx context.Context, client *api.Client, ready func(response *api.LeaderResponse) error) error {
+	var leader *api.LeaderResponse
+	var err error
+	for ctx.Err() == nil {
+		leader, err = client.Sys().Leader()
+		switch {
+		case err != nil:
+		case leader == nil:
+			err = fmt.Errorf("nil response to leader check")
+		default:
+			err = ready(leader)
+			if err == nil {
+				return nil
+			}
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	return fmt.Errorf("error checking leader: %v", err)
+}
+
+// end test helper methods
+
+// TODO: allow number of cores/servers to be configurable
+var DefaultNumCores = 1
+
+// creates a managed docker container running Vault
+func (cluster *DockerCluster) setupDockerCluster(opts *DockerClusterOptions) error {
+	if opts != nil && opts.tmpDir != "" {
+		if _, err := os.Stat(opts.tmpDir); os.IsNotExist(err) {
+			if err := os.MkdirAll(opts.tmpDir, 0700); err != nil {
+				return err
+			}
+		}
+		cluster.tmpDir = opts.tmpDir
+	} else {
+		tempDir, err := ioutil.TempDir("", "vault-test-cluster-")
+		if err != nil {
+			return err
+		}
+		cluster.tmpDir = tempDir
+	}
+	caDir := filepath.Join(cluster.tmpDir, "ca")
+	if err := os.MkdirAll(caDir, 0755); err != nil {
+		return err
+	}
+
+	var numCores int
+	if opts == nil || opts.NumCores == 0 {
+		numCores = DefaultNumCores
+	} else {
+		numCores = opts.NumCores
+	}
+
+	if opts != nil && opts.RequireClientAuth {
+		cluster.clientAuthRequired = true
+	}
+
+	cidr := "192.168.128.0/20"
```

Comment:
Can we document why this cidr is being used?  I think this is configurable in Docker, it would be good to have other OS's test this to make sure it works out of box (IE Linux).

Reply:
I ended up removing this; we no longer specify the CIDR and let Docker do this on it's own. We also generate a UUID for the network name, and clean up the network during `Teardown` in bb300a30a